### PR TITLE
[Bug]: versions 500 error when field collection is removed

### LINKED ISSUE #13749: [Bug]: versions 500 error when field collection is removed
### Expected behavior

After removing a field collection it should still be possible to view the versions of a data object.

### Actual behavior

A 500 internal server error is thrown.
Versions_500_internal_server_error

### Steps to reproduce

- Create two field collections.
  - Add a text input field to each of them.
- Create a class definition.
  - Add a field collection field and link the created field collections.
- Create a new data object of the new class definition type.
  - Add at least one of each field collection and enter a value in the text input field.
  - Save the data object.
- Remove one of the field collections from the field collection field in the class definition.
- Remove the field collection itself via the field collection editor.
- Open the latest version of the data object.

### PR DESCRIPTION
**Pull request by robertSt7**
resolves #13749

### CODE REVIEW

File: models/DataObject/Fieldcollection.php [hunk 1]
```diff
@@ -109,10 +109,16 @@ public function setFieldname($fieldname)
     public function getItemDefinitions()
     {
         $definitions = [];
-        foreach ($this->getItems() as $item) {
-            $definitions[$item->getType()] = $item->getDefinition();
+        $items = $this->getItems();
+        foreach ($items as $key => $item) {
+            if($item == null) {
```
Comment (kjkooistra-youwe) [inline models/DataObject/Fieldcollection.php]: Preferably add a space between `if` and `(` and use exact matching `===` (`"" == null` results in `true`).

Comment (robertSt7) [reply models/DataObject/Fieldcollection.php]: @kjkooistra-youwe Thanks, I have adopted the changes you requested.

**Comment by blankse** [comment]
I think this is not the right approach. The getItems method shouldn't return an array with null values. We should add a __wakeup method like in the objectbricks class which remove this values.

**Comment by blankse** [comment]
I created a PR with my solution: https://github.com/pimcore/pimcore/pull/13957

**Comment by robertSt7** [comment]
fixed by https://github.com/pimcore/pimcore/pull/13957

**Pull request closed without merging**